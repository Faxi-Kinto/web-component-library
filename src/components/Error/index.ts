import Error from './Error.component';

export default Error;
