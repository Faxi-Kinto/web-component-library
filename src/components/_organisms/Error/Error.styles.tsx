import styled from 'styled-components';
import { pxToRem, marginChildren } from '@faxi/web-css-utilities';
import { ListView } from '../../_layouts/ListView';

export type ErrorStylingProps = {
  errorMessageTitleColor?: string;
  errorMessageDescriptionColor?: string;
  errorLinksPhoneColor?: string;
};

export const Error = styled.div`
  ${ListView(pxToRem('36px'), 'center', 'center')};
  text-align: center;
  height: 100%;

  .error {
    &__logo {
      justify-content: center;
    }

    &__message {
      ${marginChildren(`0 0 ${pxToRem('15px')} 0`)};
      font-weight: 500;

      &__title {
        color: ${(props: ErrorStylingProps) =>
          props.errorMessageTitleColor || 'black'};
        font-weight: 500;
      }

      &__description {
        font-size: ${pxToRem('20px')};
        color: ${(props: ErrorStylingProps) =>
          props.errorMessageDescriptionColor || 'black'};
      }
    }

    &__links {
      ${marginChildren(`0 0 ${pxToRem('15px')} 0`)};
      font-weight: 200;

      &__nav-links {
        font-size: ${pxToRem('20px')};
      }

      &__phone {
        font-size: ${pxToRem('20px')};
        color: ${(props: ErrorStylingProps) =>
          props.errorLinksPhoneColor || 'black'};
      }

      &__message {
        font-size: ${pxToRem('20px')};
      }
    }
  }
`;
