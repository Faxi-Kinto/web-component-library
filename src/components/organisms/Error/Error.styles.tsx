import styled from 'styled-components';
import { listView } from 'styles/containers';
import { pxToRem } from 'styles/fonts';
import { marginChildren } from 'styles/spacings';

export const Error = styled.div`
  ${listView(pxToRem('36px'), 'center', 'center')};
  text-align: center;
  height: 100%;

  .error {
    &__logo {
      justify-content: center;
    }

    &__message {
      ${marginChildren(`0 0 ${pxToRem('15px')} 0`)};

      &__description {
        font-size: ${pxToRem('18px')};
      }
    }

    &__links {
      ${marginChildren(`0 0 ${pxToRem('15px')} 0`)};
      font-weight: 200;

      &__message {
        font-size: ${pxToRem('18px')};
      }
    }
  }
`;
