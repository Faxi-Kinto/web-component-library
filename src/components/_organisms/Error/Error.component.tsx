import React, { ReactElement } from 'react';
import * as Styled from './Error.styles';
import Text from '../../_atoms/Text';

export type ErrorType = '400' | '404' | '500';

export type ErrorItems = {
  description: string;
  message: string;
  jsx: ReactElement;
};

export type ErrorProps = {
  errorType: ErrorType;
  ErrorTexts?: Record<ErrorType, ErrorItems>;
  icon?: JSX.Element;
  errorMessageTitle?: string;
} & Styled.ErrorStylingProps;

const Error: React.FC<ErrorProps> = (props: ErrorProps): JSX.Element => {
  const {
    errorType = '404',
    icon,
    ErrorTexts,
    errorMessageTitle,
    errorMessageTitleColor,
    errorMessageDescriptionColor,
    errorLinksPhoneColor,
  } = props;

  return (
    <Styled.Error
      className="error"
      errorMessageTitleColor={errorMessageTitleColor}
      errorMessageDescriptionColor={errorMessageDescriptionColor}
      errorLinksPhoneColor={errorLinksPhoneColor}
    >
      {icon &&
        React.cloneElement(icon, {
          className: 'error__logo',
        })}
      <div className="error__message">
        <Text.Heading level="1" className="error__message__title">
          {errorMessageTitle}
        </Text.Heading>
        {ErrorTexts && (
          <span className="error__message__description">
            {ErrorTexts[errorType].description}
          </span>
        )}
      </div>
      {ErrorTexts && (
        <div className="error__links">
          <div className="error__links__message">
            {ErrorTexts[errorType].message}
          </div>
          {ErrorTexts[errorType].jsx}
        </div>
      )}
    </Styled.Error>
  );
};

Error.defaultProps = {
  errorType: '404',
};

export default Error;
