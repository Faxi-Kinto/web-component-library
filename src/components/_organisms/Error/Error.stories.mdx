import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { withKnobs, select } from '@storybook/addon-knobs';
import Text from '../../_atoms/Text';
import NavigationLinks, {
  NavLink,
} from '../../_molecules/NavigationLinks/NavigationLinks.component';
import Error, { ErrorItems, ErrorType } from './Error.component';

<Meta title="Core|Organisms/Error" />

export const errorLinks = [
  { name: 'Home', route: '' },
  { name: 'About', route: '' },
  { name: 'Enterprise', route: '' },
  { name: 'FAQ', route: '' },
];

export const ErrorTexts = {
  '400': {
    description: 'Network error',
    message: 'Network error message',
    jsx: (
      <Text.Link className="error__links__phone" href="tel:+111111111">
        (+1) 11111111
      </Text.Link>
    ),
  },
  '404': {
    description: 'Not found',
    message: 'Not found message',
    jsx: <NavigationLinks links={errorLinks} vertical />,
  },
  '500': {
    description: 'Server error',
    message: 'Server error message',
    jsx: (
      <Text.Link className="error__links__phone" href="tel:+111111111">
        (+1) 11111111
      </Text.Link>
    ),
  },
};

export const label = 'ErrorTypes';
export const options = {
  400: '400',
  404: '404',
  500: '500',
};
export const defaultValue = '404';
export const groupId = 'GROUP-ID1';

# Error Component

## Implementation

#### Default

<Preview>
  <Story name="default" decorators={[withKnobs]}>
    <Error
      errorType={select(label, options, defaultValue, groupId)}
      ErrorTexts={ErrorTexts}
      errorMessageTitle="An error occured"
    />
  </Story>
</Preview>

## Props

<Props of={Error} />
