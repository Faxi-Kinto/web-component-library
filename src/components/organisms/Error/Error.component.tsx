import React, { ReactElement } from 'react';
import * as Styled from './Error.styles';
import Text from '../../atoms/Text';
import SubNavigation from '../../molecules/SubNavigation';
import Icon from '../../atoms/Icon';
import { NavLink } from '../../molecules/SubNavigation';
import { theme } from '../../../config';
import { pxToRem } from '../../../styles/basics';

const errorLinks: NavLink[] = [
  { name: 'Home', route: '' },
  { name: 'About', route: '' },
  { name: 'Enterprise', route: '' },
  { name: 'FAQ', route: '' },
];

export type ErrorType = '400' | '404' | '500';

type ErrorItems = {
  description: string;
  message: string;
  jsx: ReactElement;
};

export type ErrorProps = {
  errorType: ErrorType;
};

const ErrorTexts: Record<ErrorType, ErrorItems> = {
  '400': {
    description: `It seems like there's a network error.`,
    message: 'Please try again later or call our support line:',
    jsx: <Text.Link href={'tel:123-456'}>+40 123 456 789</Text.Link>,
  },
  '404': {
    description: `We can't seem to find the page you're looking for.`,
    message: 'Here are some helpful links instead',
    jsx: <SubNavigation links={errorLinks} vertical colorType="dark" />,
  },
  '500': {
    description: `It seems like the server is down.`,
    message: 'Please try again later or call our support line:',
    jsx: <Text.Link href={'tel:123-456'}>+40 123 456 789</Text.Link>,
  },
};

const Error: React.FC<ErrorProps> = (props: ErrorProps): JSX.Element => {
  const { errorType } = props;

  return (
    <Styled.Error className="error">
      <Icon
        name="frown"
        className="error__logo"
        size={pxToRem('100px')}
        color={theme.pallet.LIGHT_BLUE}
      />
      <div className="error__message">
        <Text.Heading level="1">Oops!</Text.Heading>
        <span className="error__message__description">
          {ErrorTexts[errorType].description}
        </span>
      </div>
      <div className="error__links">
        <div className="error__links__message">
          {ErrorTexts[errorType].message}
        </div>
        {ErrorTexts[errorType].jsx}
      </div>
    </Styled.Error>
  );
};

export default Error;
