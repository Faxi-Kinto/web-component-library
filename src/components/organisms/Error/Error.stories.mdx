import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { theme } from '../../../config';

import Error from './';

<Meta title="Core|Organisms/Error" />

# Error Component

## Implementation

#### Default

<Preview>
  <Story name="default">
    <Error />
  </Story>
</Preview>

## Props

<Props of={Error} />
