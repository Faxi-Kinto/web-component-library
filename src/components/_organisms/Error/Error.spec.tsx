import React from 'react';
import renderer from 'react-test-renderer';
import Error from './Error.component';

test('Error renders correctly', () => {
  const component = renderer.create(<Error errorType="404" />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
