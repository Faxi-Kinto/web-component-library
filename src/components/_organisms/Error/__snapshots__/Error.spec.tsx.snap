// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  text-align: center;
  height: 100%;
}

.c0 > :first-child {
  max-width: 100%;
  width: 100%;
}

.c0 > * {
  margin: 0 0 2.25rem 0;
}

.c0 > :last-child {
  margin: 0;
}

.c0 .error__logo {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .error__message {
  font-weight: 500;
}

.c0 .error__message > * {
  margin: 0 0 0.9375rem 0;
}

.c0 .error__message > :last-child {
  margin: 0;
}

.c0 .error__message__title {
  color: black;
  font-weight: 500;
}

.c0 .error__message__description {
  font-size: 1.25rem;
  color: black;
}

.c0 .error__links {
  font-weight: 200;
}

.c0 .error__links > * {
  margin: 0 0 0.9375rem 0;
}

.c0 .error__links > :last-child {
  margin: 0;
}

.c0 .error__links__nav-links {
  font-size: 1.25rem;
}

.c0 .error__links__phone {
  font-size: 1.25rem;
  color: black;
}

.c0 .error__links__message {
  font-size: 1.25rem;
}

.c1 {
  padding: 0;
  margin: 0;
  font-size: 2.125rem;
  line-height: 2.9375rem;
  font-weight: bold;
  color: black;
}

<div
  className="c0 error"
>
  <div
    className="error__message"
  >
    <h1
      className="c1 error__message__title"
      color="black"
    />
  </div>
</div>
`;
